### PR TITLE
IN: Profile details integration

### LINKED ISSUE #80: IN: Profile Details

### PR DESCRIPTION
**Pull request by Habib-Ahmad**
### What this PR does (required):
- When a user is clicked on, from the dashboard, the user's details are displayed on the profile details page.
- It makes use of the getProfileById API
- Allows a user create a booking with the booking form
- closes #80 

### Any information needed to test this feature (required):
- User needs to be logged in

### CODE REVIEW

File: client/src/pages/ProfileDetail/ProfileDetail.tsx [hunk 1]
```diff
@@ -1,61 +1,98 @@
...
+
+  const getProfile = useCallback(async () => {
+    setLoading(true);
+    try {
+      const userId = match.params.userId;
+      const response = await getProfileFromId(userId);
+      setProfile(response);
+      setLoading(false);
+    } catch (error) {
+      console.log(error);
```
Comment (sundayezeilo) [inline client/src/pages/ProfileDetail/ProfileDetail.tsx:48]: You could use `console.error` for this log

File: server/controllers/profile.js [hunk 2]
```diff
@@ -38,4 +36,22 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
       profile: profile,
     },
   });
-});
\ No newline at end of file
+});
+
+// @route GET /profile/loadFromId
+// @desc Get user profile data
+// @access Private
```
Comment (sundayezeilo) [inline server/controllers/profile.js:43]: Good documentation here

File: server/controllers/profile.js [hunk 3]
```diff
@@ -38,4 +36,22 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
...
+  if (!profile) {
+    res.status(404);
+    throw new Error("Profile does not exist");
+  }
+
+  res.status(200).json({
+    data: {
+      profile: profile,
+    },
+  });
```
Comment (sundayezeilo) [inline server/controllers/profile.js:56]: :+1:

**Review by sundayezeilo — COMMENTED** [review]
Nice work with integration :+1: Please do link your ticket next time.

**Comment by moffatethan** [comment]
If you could make the booking form create a new request, that would be great!

**Review by moffatethan — APPROVED** [review]

**Pull request merged**